integration-tests/src/test/java/oracle/weblogic/kubernetes/ItRetryImprovements.java

```diff
@@ -176,12 +188,132 @@ void testRetryStoppedAfterIssueFixed() {
+   * actions to fix the problem.
+   * Verify that retry resume after the issue is fixed and the domain starts successfully.
+   */
+  @Test
+  @DisplayName("Create a domain duplicate server names. Verify that retry stopped and handles FATAL error as designed.")
+  void testRetryOccursAsExpectedAndThrowFatalFailures() {
+    int replicaCount = 2;
+    String duplicateServerName = "managed-server1";
+    // verify that the operator stops retrying immediately when a FATAL error occurs and clear message is logged.
+    DomainResource domain = createDomainResourceForRetryTest(FAILURE_RETRY_LIMIT_MINUTES, replicaCount, true);
+
+    // create an invalid domain resource with duplicate server names
+    domain.getSpec().addManagedServersItem(new ManagedServer().serverName(duplicateServerName));
+    domain.getSpec().addManagedServersItem(new ManagedServer().serverName(duplicateServerName));
+    createDomainForRetryTest(domain);
+
+    String fatalDomainInvalidErrorRegex =
+        new StringBuffer(".*FatalDomainInvalidError.*More than one.*spec.managedServers.*")
+          .append(duplicateServerName)
+          .append(".*").toString();
+
+    // verify that FatalDomainInvalidError message found in domain status message
+    testUntil(() -> findStringInDomainStatusMessage(domainNamespace, domainUid, fatalDomainInvalidErrorRegex),
+        logger, "FatalDomainInvalidError is found in domain status message");
+
+    // verify that FatalDomainInvalidError message found in Operator log
+    testUntil(() -> findStringInOperatorLog(opNamespace, fatalDomainInvalidErrorRegex),
+        logger, "FatalDomainInvalidError is found in Operator log");
+
+    // patch domain CR to delete duplicate server and have Operator retry
+    int introspectVersion = 0;
+    if (domain.getSpec().getIntrospectVersion() != null) {
+      introspectVersion = Integer.parseInt(domain.getSpec().getIntrospectVersion()) + 1;
+    }
+
+    StringBuffer patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"remove\",")
+        .append(" \"path\": \"/spec/managedServers/0\"},")
+        .append("{\"op\": \"replace\", \"path\": \"/spec/introspectVersion\", \"value\": \"")
+        .append(introspectVersion)
+        .append("\"}]");
+    logger.info("PatchStr for : {0}", patchStr.toString());
+
+    boolean cmPatched = patchDomainResource(domainUid, domainNamespace, patchStr);
+    assertTrue(cmPatched, "Patch domain CR to delete duplicate server failed");
+
+    verifyDomainExistsAndServerStarted(replicaCount);
+  }
+
+  /**
+   * Create a domain-in-image domain with replica count = 6 that exceeds the maximum cluster size
+   * of 5. The domain should start but with WARNING message.
+   * Also log a clear message in Operator logs and the domain with the cause.
+   */
+  @Test
+  @DisplayName("Create a domain duplicate server names. Verify that retry stopped and handles Waring as designed.")
```

Comment:
Waring --> Warning
Create a domain duplicate server names --> Create a domain with duplicate server names
Test description is same for both cases  ... Create a domain with duplicate server names

Reply:
Changed